### PR TITLE
feat: DEV-3283: New comment behaviour for Submit/Update/Skip/Accept/Reject buttons

### PR DESCRIPTION
**Pull request by yyassi-heartex**

### CODE REVIEW

File: src/components/TopBar/Controls.js [hunk 1]
```diff
@@ -6,9 +6,7 @@ import { isDefined } from "../../utils/utilities";
 import { IconBan } from "../../assets/icons";
 
 import "./Controls.styl";
-import { useCallback, useMemo, useState } from "react";
-import { Dropdown } from "../../common/Dropdown/DropdownComponent";
-import { FF_DEV_1593, FF_DEV_2186, FF_DEV_2458, isFF } from "../../utils/feature-flags";
```
[the remark this answers is not in the document]
Comment (yyassi-heartex) [reply src/components/TopBar/Controls.js:11]: yes thats intended - these FFs were being used but now that we have the toggles in the review and annotation settings they are no longer needed. I also updated https://heartex.atlassian.net/browse/DEV-3283 to make sure we deprecate/remove those FFs from LD

File: src/components/Comments/CommentForm.tsx [hunk 2]
```diff
@@ -25,8 +26,10 @@ export const CommentForm: FC<CommentFormProps> = observer(({
   maxRows = 4,
 }) => {
   const formRef = useRef<HTMLFormElement>(null);
-  const actionRef = useRef<{ update?: (text?: string) => void }>({});
-
+  const actionRef = useRef<{ update?: (text?: string) => void, ele?: RefObject<HTMLTextAreaElement> }>({});
```
Comment (bmartel) [inline src/components/Comments/CommentForm.tsx:29]:
```suggestion
  const actionRef = useRef<{ update?: (text?: string) => void, el?: RefObject<HTMLTextAreaElement> }>({});
```
It is really nitpicky but conventional shortened form of element is `el`, can we change that?

File: src/common/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -11,8 +11,10 @@ export type TextAreaProps = {
   onSubmit?: () => void|Promise<void>,
   onChange?: (value: string) => void,
   onInput?: (value: string) => void,
+  onFocus?: (e: FocusEvent) => void,
+  onBlur?: (e: FocusEvent) => void,
   ref?: MutableRefObject<HTMLTextAreaElement>,
-  actionRef?: MutableRefObject<{ update?: (text?: string) => void }>,
+  actionRef?: MutableRefObject<{ update?: (text?: string) => void, ele?: RefObject<HTMLTextAreaElement> }>,
```
Comment (bmartel) [inline src/common/TextArea/TextArea.tsx:17]:
```suggestion
  actionRef?: MutableRefObject<{ update?: (text?: string) => void, el?: RefObject<HTMLTextAreaElement> }>,
```

File: src/common/TextArea/TextArea.tsx [hunk 4]
```diff
@@ -98,6 +100,7 @@ export const TextArea: FC<TextAreaProps> = ({
         textAreaRef.current.value = text;
         resizeTextArea();
       },
+      ele: textAreaRef,
```
Comment (bmartel) [inline src/common/TextArea/TextArea.tsx:103]:
```suggestion
      el: textAreaRef,
```

File: src/components/TopBar/Controls.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yyassi-heartex) [reply src/components/TopBar/Controls.js:11]: i see the declaration for FF_DEV_2458 is still there so i just went ahead and removed it

File: src/components/Comments/CommentForm.tsx [hunk 5]
```diff
@@ -51,6 +56,17 @@ export const CommentForm: FC<CommentFormProps> = observer(({
...
   }, [commentStore]);
 
+  
+  useEffect(() => {
+    commentStore.setAddedCommentThisSession(false);
+    clearTooltipMessage();
+  }, []);
+
+  useEffect(() => {
+    commentStore.setInputRef(actionRef.current.ele);
```
Comment (bmartel) [inline src/components/Comments/CommentForm.tsx:66]:
```suggestion
    commentStore.setInputRef(actionRef.current.el);
```

**Review by bmartel — CHANGES_REQUESTED** [review]
Looks great, one really small change requested but the code is sound.

**Review by bmartel — APPROVED** [review]

**Pull request merged**